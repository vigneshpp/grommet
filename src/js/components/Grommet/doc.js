import { describe, PropTypes } from 'react-desc';

import { backgroundDoc } from '../../utils/prop-types';
import { getAvailableAtBadge } from '../../utils/mixins';

export const doc = Grommet => {
  const DocumentedGrommet = describe(Grommet)
    .availableAt(getAvailableAtBadge('Grommet', 'Utilities'))
    .description('The top level Grommet container.')
    .usage(
      `import { Grommet } from 'grommet';
<Grommet>...</Grommet>`,
    )
    .intrinsicElement('div');

  DocumentedGrommet.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description(
      'Layout direction for right to left contexts',
    ),
    full: PropTypes.bool
      .description('Whether to take the whole viewport.')
      .defaultValue(false),
    plain: PropTypes.bool
      .description(
        `Whether or not Grommet should apply a global font-family, font-size,
        and line-height.`,
      )
      .defaultValue(false),
    cssVars: PropTypes.bool
      .description('Whether to expose the css variables.')
      .defaultValue(false),
    theme: PropTypes.object.description(
      'Custom styles for Grommet app component.',
    ),
    themeMode: PropTypes.oneOf(['dark', 'light']).description(
      `Dark vs. light theme variation. Default is unspecified and left to
      theme.`,
    ),
    userAgent: PropTypes.string.description(
      `User agent used to detect the device width for setting the initial
      breakpoint.`,
    ),
    containerTarget: PropTypes.object.description(
      `The node where Drop and Layer containers are inserted. Defaults to
      document.body which is almost always the right choice. This is used
      for less common cases like rendering within an internal node (e.g.
      shadow root).`,
    ),
  };

  return DocumentedGrommet;
};

export const themeDoc = {
  'grommet.extend': {
    description: 'Any additional style for Grommet.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
};
