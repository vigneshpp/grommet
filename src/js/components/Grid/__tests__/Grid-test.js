import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Grommet } from '../../Grommet';
import { Grid } from '..';

describe('Grid', () => {
  test('renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('a11yTitle renders', () => {
    const { container, getByLabelText } = render(
      <Grommet>
        <Grid a11yTitle="My Grid" />
      </Grommet>,
    );
    const gridWithLabel = getByLabelText('My Grid');
    expect(gridWithLabel).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('rows renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid rows={[['small', 'medium'], 'large', 'medium']} />
        <Grid rows={['small', 'large', 'medium']} />
        <Grid rows="small" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('rows renders with warning', () => {
    console.warn = jest.fn();
    const warnSpy = jest.spyOn(console, 'warn');
    const component = renderer.create(
      <Grommet>
        <Grid rows={['flex']} fill={false} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(warnSpy).toHaveBeenCalledWith(
      'Grid needs `fill` when using fractional row sizes',
    );
  });

  test('columns renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid columns={['1/2', '2/4']} />
        <Grid columns={['1/3', '2/3']} />
        <Grid columns={['1/4', '3/4']} />
        <Grid columns={[['1/2', '2/4'], '1/4', '3/4']} />
        <Grid columns="small" />
        <Grid columns="flex" />
        <Grid columns={{ count: 'fit', size: 'small' }} />
        <Grid columns={{ count: 'fill', size: ['small', 'medium'] }} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('areas renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid
          rows={['xxsmall', 'medium', 'xsmall']}
          columns={['3/4', '1/4']}
          areas={[
            { name: 'header', start: [0, 0], end: [0, 1] },
            { name: 'main', start: [1, 0], end: [1, 0] },
            { name: 'sidebar', start: [1, 1], end: [1, 1] },
            { name: 'footer', start: [2, 0], end: [2, 1] },
          ]}
        />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('areas renders with warning and throws error', () => {
    console.error = jest.fn();
    console.warn = jest.fn();
    const warnSpy = jest.spyOn(console, 'warn');
    expect(() => {
      renderer.create(
        <Grommet>
          <Grid
            rows={['xxsmall', 'medium', 'xsmall']}
            columns="small"
            areas={[
              { name: 'header', start: [0, 0], end: [0, 1] },
              { name: 'main', start: [1, 0], end: [1, 0] },
              { name: 'sidebar', start: [1, 1], end: [1, 1] },
              { name: 'footer', start: [2, 0], end: [2, 1] },
            ]}
          />
        </Grommet>,
      );
    }).toThrow('props.columns.map is not a function');
    expect(warnSpy).toHaveBeenCalledWith(
      'Grid `areas` requires `rows` and `columns` to be arrays.',
    );
  });

  test('areas renders when given an array of string arrays', () => {
    const component = renderer.create(
      <Grommet>
        <Grid
          rows={['xxsmall', 'medium', 'xsmall']}
          columns={['3/4', '1/4']}
          areas={[
            ['header', 'header'],
            ['sidebar', 'main'],
            ['footer', 'footer'],
          ]}
        />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('justify renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid justify="start" />
        <Grid justify="center" />
        <Grid justify="end" />
        <Grid justify="stretch" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('align renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid align="start" />
        <Grid align="center" />
        <Grid align="end" />
        <Grid align="stretch" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('justifyContent renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid justifyContent="start" />
        <Grid justifyContent="center" />
        <Grid justifyContent="between" />
        <Grid justifyContent="around" />
        <Grid justifyContent="end" />
        <Grid justifyContent="stretch" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('alignContent renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid alignContent="start" />
        <Grid alignContent="center" />
        <Grid alignContent="between" />
        <Grid alignContent="around" />
        <Grid alignContent="end" />
        <Grid alignContent="stretch" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('gap renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid gap="small" />
        <Grid gap="medium" />
        <Grid gap="large" />
        <Grid gap={{ row: 'small' }} />
        <Grid gap={{ row: 'medium' }} />
        <Grid gap={{ row: 'large' }} />
        <Grid gap={{ column: 'small' }} />
        <Grid gap={{ column: 'medium' }} />
        <Grid gap={{ column: 'large' }} />
        <Grid gap={{ row: 'small', column: 'medium' }} />
        <Grid gap={{ test: 'test' }} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('fill renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid fill />
        <Grid fill={false} />
        <Grid fill="horizontal" />
        <Grid fill="vertical" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('responsive', () => {
    const component = renderer.create(
      <Grommet>
        <Grid responsive />
        <Grid responsive={false} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('as renders', () => {
    const component = renderer.create(
      <Grommet>
        <Grid as="article" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('proxies tag', () => {
    const tagComponent = renderer.create(
      <Grommet>
        <Grid tag="article" />
      </Grommet>,
    );
    const asComponent = renderer.create(
      <Grommet>
        <Grid as="article" />
      </Grommet>,
    );
    expect(tagComponent.toJSON()).toEqual(asComponent.toJSON());
  });

  test('pad', () => {
    const component = renderer.create(
      <Grommet>
        <Grid pad="small" />
        <Grid pad="medium" />
        <Grid pad="large" />
        <Grid pad={{ horizontal: 'small' }} />
        <Grid pad={{ vertical: 'small' }} />
        <Grid pad={{ bottom: 'small' }} />
        <Grid pad={{ left: 'small' }} />
        <Grid pad={{ right: 'small' }} />
        <Grid pad={{ start: 'small' }} />
        <Grid pad={{ end: 'small' }} />
        <Grid pad={{ top: 'small' }} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('border', () => {
    const component = renderer.create(
      <Grommet>
        <Grid border="all" />
        <Grid border="horizontal" />
        <Grid border="vertical" />
        <Grid border="top" />
        <Grid border="left" />
        <Grid border="bottom" />
        <Grid border="right" />
        <Grid border={{ color: 'accent-1' }} />
        <Grid border={{ side: 'all' }} />
        <Grid border={{ size: 'xsmall' }} />
        <Grid border={{ size: 'small' }} />
        <Grid border={{ size: 'medium' }} />
        <Grid border={{ size: 'large' }} />
        <Grid border={{ size: 'xlarge' }} />
        <Grid border={{ style: 'dotted' }} />
        <Grid border={{ style: 'double' }} />
        <Grid border={{ style: 'dashed' }} />
        <Grid
          border={[
            { side: 'top', color: 'accent-1', size: 'medium', style: 'dotted' },
            { side: 'left', color: 'accent-2', size: 'large', style: 'dashed' },
          ]}
        />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
