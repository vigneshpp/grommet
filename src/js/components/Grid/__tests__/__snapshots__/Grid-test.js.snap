// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid a11yTitle renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
}

<div>
  <div
    class="c0"
  >
    <div
      aria-label="My Grid"
      class="c1"
    />
  </div>
</div>
`;

exports[`Grid align renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid alignContent renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
</div>
`;

exports[`Grid areas renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "header main footer" "header sidebar footer" ". .";
  grid-template-columns: 75% 25%;
  grid-template-rows: 48px 384px 96px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Grid areas renders when given an array of string arrays 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "header header" "sidebar main" "footer footer";
  grid-template-columns: 75% 25%;
  grid-template-rows: 48px 384px 96px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Grid as renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <article
    className="c1"
  />
</div>
`;

exports[`Grid border 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  border: solid 1px rgba(0,0,0,0.33);
}

.c2 {
  display: grid;
  box-sizing: border-box;
  border-top: solid 1px rgba(0,0,0,0.33);
  border-bottom: solid 1px rgba(0,0,0,0.33);
}

.c3 {
  display: grid;
  box-sizing: border-box;
  border-left: solid 1px rgba(0,0,0,0.33);
  border-right: solid 1px rgba(0,0,0,0.33);
}

.c4 {
  display: grid;
  box-sizing: border-box;
  border-top: solid 1px rgba(0,0,0,0.33);
}

.c5 {
  display: grid;
  box-sizing: border-box;
  border-left: solid 1px rgba(0,0,0,0.33);
}

.c6 {
  display: grid;
  box-sizing: border-box;
  border-bottom: solid 1px rgba(0,0,0,0.33);
}

.c7 {
  display: grid;
  box-sizing: border-box;
  border-right: solid 1px rgba(0,0,0,0.33);
}

.c8 {
  display: grid;
  box-sizing: border-box;
  border: solid 1px #6FFFB0;
}

.c9 {
  display: grid;
  box-sizing: border-box;
  border: solid 2px rgba(0,0,0,0.33);
}

.c10 {
  display: grid;
  box-sizing: border-box;
  border: solid 4px rgba(0,0,0,0.33);
}

.c11 {
  display: grid;
  box-sizing: border-box;
  border: solid 12px rgba(0,0,0,0.33);
}

.c12 {
  display: grid;
  box-sizing: border-box;
  border: solid 24px rgba(0,0,0,0.33);
}

.c13 {
  display: grid;
  box-sizing: border-box;
  border: dotted 1px rgba(0,0,0,0.33);
}

.c14 {
  display: grid;
  box-sizing: border-box;
  border: double 1px rgba(0,0,0,0.33);
}

.c15 {
  display: grid;
  box-sizing: border-box;
  border: dashed 1px rgba(0,0,0,0.33);
}

.c16 {
  display: grid;
  box-sizing: border-box;
  border-top: dotted 4px #6FFFB0;
  border-left: dashed 12px #FD6FFF;
}

@media only screen and (max-width:768px) {
  .c1 {
    border: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    border-top: solid 1px rgba(0,0,0,0.33);
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-left: solid 1px rgba(0,0,0,0.33);
    border-right: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    border-top: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    border-left: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    border-right: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    border: solid 1px #6FFFB0;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    border: solid 2px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c10 {
    border: solid 4px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    border: solid 6px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c12 {
    border: solid 12px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    border: dotted 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c14 {
    border: double 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    border: dashed 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    border-top: dotted 4px #6FFFB0;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    border-left: dashed 6px #FD6FFF;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c1"
  />
  <div
    className="c1"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
  <div
    className="c12"
  />
  <div
    className="c13"
  />
  <div
    className="c14"
  />
  <div
    className="c15"
  />
  <div
    className="c16"
  />
</div>
`;

exports[`Grid columns renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 50% 50%;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 33.33% 66.66%;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 25% 75%;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: minmax(50%,50%) 25% 75%;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(min(192px,100%),1fr) );
}

.c6 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,1fr );
}

.c7 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fit,minmax(min(192px,100%),1fr) );
}

.c8 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(min(192px,100%),384px) );
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
</div>
`;

exports[`Grid fill renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

.c2 {
  display: grid;
  box-sizing: border-box;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  width: 100%;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  height: 100%;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid gap renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 12px 12px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 24px 24px;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 48px 48px;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 12px;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 24px;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 48px;
}

.c7 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 12px;
}

.c8 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 24px;
}

.c9 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 48px;
}

.c10 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 12px;
  grid-column-gap: 24px;
}

.c11 {
  display: grid;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
</div>
`;

exports[`Grid justify renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  justify-items: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  justify-items: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  justify-items: flex-end;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  justify-items: stretch;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid justifyContent renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
</div>
`;

exports[`Grid pad 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  padding: 12px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  padding: 24px;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  padding: 48px;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  padding-left: 12px;
  padding-right: 12px;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  padding-bottom: 12px;
}

.c7 {
  display: grid;
  box-sizing: border-box;
  padding-left: 12px;
}

.c8 {
  display: grid;
  box-sizing: border-box;
  padding-right: 12px;
}

.c9 {
  display: grid;
  box-sizing: border-box;
  padding-inline-start: 12px;
}

.c10 {
  display: grid;
  box-sizing: border-box;
  padding-inline-end: 12px;
}

.c11 {
  display: grid;
  box-sizing: border-box;
  padding-top: 12px;
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding-left: 6px;
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding-top: 6px;
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    padding-left: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding-inline-start: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c10 {
    padding-inline-end: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    padding-top: 6px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
</div>
`;

exports[`Grid renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Grid responsive 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c1"
  />
</div>
`;

exports[`Grid rows renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-rows: minmax(192px,384px) 768px 384px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-template-rows: 192px 768px 384px;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  grid-auto-rows: 192px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
</div>
`;

exports[`Grid rows renders with warning 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-rows: 1fr;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
